### DISCUSSION TITLE
Could not process [path] file, due to: Undefined constant

### DISCUSSION THREAD

**Discussion by escopecz** [rectorphp/rector#7212] [Q&A] [answered]
Dear Rector community,

we are preparing support for PHP 8.0 in Mautic and the Rector [v0.12.22](https://github.com/mautic/mautic/blob/4.x/composer.lock#L15850) is [passing on PHP 7.4](https://github.com/mautic/mautic/runs/6753334604?check_suite_focus=true) but [failing on PHP 8.0](https://github.com/mautic/mautic/runs/6753334693?check_suite_focus=true) with error message:

```
Could not process "app/bundles/EmailBundle/EventListener/ReportSubscriber.php" file, due to: "Undefined constant "Mautic\EmailBundle\EventListener\MAUTIC_TABLE_PREFIX". On line: 72
```

We cannot understand why when looking at the file on that line:

https://github.com/mautic/mautic/blob/4.3/app/bundles/EmailBundle/EventListener/ReportSubscriber.php#L72

This "fix" made the CI pass but it's not optimal:

https://github.com/mautic/mautic/pull/11217/commits/e3768d29cc1854a52cac71cac3272641eea9bb64

Can anyone give us some pointers about why is it failing and what would be the correct fix?

**Comment by TomasVotruba** [comment]
Hi Honza,

there have been many new releases of PHPStan 1.7.* that handles magical autoloading in different way.
Make sure to upgrade to the latest PHPStan 1.7.10 and Rector 0.13.4 first.

**Comment by samsonasik** [comment, marked as answer]
You can define your own bootstrap file instead of define it in the rector.php, so you can configure:

```php
    // do you need to include constants, class aliases or custom autoloader? files listed will be executed
    $rectorConfig->bootstrapFiles([
        __DIR__ . '/bootstrap.php',
    ]);
```

ref https://github.com/rectorphp/rector/blob/main/docs/static_reflection_and_autoload.md#include-files
